propulsion-projector/Program.fs

```diff
@@ -230,6 +230,40 @@ module Args =
                 seconds x.Timeout, x.Retries, seconds x.MaxRetryWaitTime)
             let connector = Equinox.Cosmos.Connector(x.Timeout, x.Retries, x.MaxRetryWaitTime, Log.Logger, mode=x.Mode)
             discovery, x.Database, x.Container, connector
+#endif
+//#if sss
+    /// TODO: add DB connectors other than MsSql
+    type [<NoEquality; NoComparison>] SqlStreamStoreSourceParameters =
+        | [<AltCommandLine "-t"; Unique>]   Tail of intervalS: float
+        | [<AltCommandLine "-m"; Unique>]   BatchSize of int
+        | [<AltCommandLine "-c"; Unique>]   Connection of string
+        | [<AltCommandLine "-s">]           Schema of string
+        | [<AltCommandLine "-p"; Unique>]   Credentials of string
+        | [<Unique>]                        Checkpoints of string
+        interface IArgParserTemplate with
+            member a.Usage = a |> function
+                | Tail _ ->                 "Polling interval in Seconds. Default: 1"
+                | BatchSize _ ->            "Maximum events to request from feed. Default: 512"
+                | Connection _ ->           "Connection string for SqlStreamStore db. Optional if SQLSTREAMSTORE_CONNECTION specified"
+                | Schema _ ->               "Database schema name"
+                | Credentials _ ->          "Credentials string for SqlStreamStore db (Part of connection string, but NOT logged). Optional if SQLSTREAMSTORE_CREDENTIALS specified"
+                | Checkpoints _ ->          "Connection string for Checkpoints sql db. Optional if SQLSTREAMSTORE_CHECKPOINTS_CONNECTION specified. Default: same as `Connection`"
+    and SqlStreamStoreSourceArguments(a : ParseResults<SqlStreamStoreSourceParameters>) =
+        member __.TailInterval =            a.GetResult(Tail, 1.) |> TimeSpan.FromSeconds
+        member __.MaxBatchSize =            a.GetResult(BatchSize, 512)
+        member __.StoreConnectionString =   a.TryGetResult Connection |> defaultWithEnvVar "SQLSTREAMSTORE_CONNECTION" "Connection"
+        member __.StoreCredentialsString =  a.TryGetResult Connection |> Option.orElseWith (fun () -> EnvVar.tryGet "SQLSTREAMSTORE_CREDENTIALS")
+        member __.Schema =                  a.GetResult(Schema, null)
+        member __.CheckpointsConnectionString =
+            a.TryGetResult Checkpoints
+            |> Option.orElseWith (fun () -> EnvVar.tryGet "SQLSTREAMSTORE_CHECKPOINTS_CONNECTION")
+            |> Option.defaultValue __.StoreConnectionString
```

Comment:
The default should include Credentials here. All comes back to line 250.

Reply:
makes sense...